src/PlotlyPanel.tsx

```diff
@@ -169,54 +172,80 @@ export const PlotlyPanel: React.FC<Props> = (props) => {
 const setDataFrameId = (frame: DataFrame) => {
   if (frame.meta?.custom?.id) {
     return;
-  } else if (frame.meta?.custom) {
-    frame.meta.custom['id'] = getGuid();
-  } else if (frame.meta) {
-    frame.meta.custom = { id: getGuid() };
-  } else {
-    frame.meta = { custom: { id: getGuid() } };
   }
+  _.set(frame, 'meta.custom.id', _.uniqueId());
 };
 
-// Pull out user-selected fields from a DataFrame.
-// If the options aren't set or the fields don't exist, try to pick the first
-// available fields, prioritizing time fields on the X axis.
-const getFields = (frame: DataFrame, props: Props) => {
-  let xField = frame.fields.find((field) => field.name === props.options.xAxis.field);
-  if (!xField) {
-    xField = frame.fields.find((field) => field.type === FieldType.time);
-    if (!xField) {
-      const y = props.options.yAxis.fields || [];
-      xField = frame.fields.find((field) => !y.includes(field.name));
+const getXFields = (frames: DataFrame[], selectedField?: string) => {
+  if (frames.length === 0) {
+    throw Error('No data');
+  } else if (frames.some(frame => frame.fields.length < 2)) {
+    throw Error(`One or more series does not contain at least two fields.`)
```

Comment:
I actually looked this up last week because I wasn't sure if "one or more" is supposed to be plural or not. Turns out, there's no accepted answer! For whatever reason, 'does' sounded better to me. Maybe I can ask our tech writer to be a tie breaker 😄

https://english.stackexchange.com/questions/13284/which-is-correct-one-or-more-is-or-one-or-more-are#:~:text=There%20is%20no%20general%20agreement,single%20case%20or%20general%20principle.